c_src/dpiStmt_nif.c

```diff
@@ -297,6 +297,104 @@ DPI_NIF_FUN(stmt_close)
+        dpiStmt_getInfo(stmtRes->stmt, &info),
+        stmtRes);
+
+    ERL_NIF_TERM map = enif_make_new_map(env);
+
+    enif_make_map_put(
+        env, map, enif_make_atom(env, "isDDL"),
+        info.isDDL? ATOM_TRUE : ATOM_FALSE,
+        &map);
+    enif_make_map_put(
+        env, map, enif_make_atom(env, "isDML"),
+        info.isDML? ATOM_TRUE : ATOM_FALSE,
+        &map);
+    enif_make_map_put(
+        env, map, enif_make_atom(env, "isPLSQL"),
+        info.isPLSQL? ATOM_TRUE : ATOM_FALSE,
+        &map);
+    enif_make_map_put(
+        env, map, enif_make_atom(env, "isQuery"),
+        info.isQuery? ATOM_TRUE : ATOM_FALSE,
+        &map);
+    enif_make_map_put(
+        env, map, enif_make_atom(env, "isReturning"),
+        info.isReturning? ATOM_TRUE : ATOM_FALSE,
+        &map);
+
+        ERL_NIF_TERM type;
+
+        switch (info.statementType){
+            case DPI_STMT_TYPE_UNKNOWN:
+                type = enif_make_atom(env, "DPI_STMT_TYPE_UNKNOWN");
+                break;
+            case DPI_STMT_TYPE_SELECT:
+                type = enif_make_atom(env, "DPI_STMT_TYPE_SELECT");
+                break;
+            case DPI_STMT_TYPE_UPDATE:
+                type = enif_make_atom(env, "DPI_STMT_TYPE_UPDATE");
+                break;
+            case DPI_STMT_TYPE_DELETE:
+                type = enif_make_atom(env, "DPI_STMT_TYPE_DELETE");
+                break;
+            case DPI_STMT_TYPE_INSERT:
+                type = enif_make_atom(env, "DPI_STMT_TYPE_INSERT");
+                break;
+            case DPI_STMT_TYPE_CREATE:
+                type = enif_make_atom(env, "DPI_STMT_TYPE_CREATE");
+                break;
+            case DPI_STMT_TYPE_DROP:
+                type = enif_make_atom(env, "DPI_STMT_TYPE_DROP");
+                break;
+            case DPI_STMT_TYPE_ALTER:
+                type = enif_make_atom(env, "DPI_STMT_TYPE_ALTER");
+                break;
+            case DPI_STMT_TYPE_BEGIN:
+                type = enif_make_atom(env, "DPI_STMT_TYPE_BEGIN");
+                break;
+            case DPI_STMT_TYPE_DECLARE:
+                type = enif_make_atom(env, "DPI_STMT_TYPE_DECLARE");
+                break;
+            case DPI_STMT_TYPE_CALL:
+                type = enif_make_atom(env, "DPI_STMT_TYPE_CALL");
+                break;
+            case DPI_STMT_TYPE_MERGE:
+                type = enif_make_atom(env, "DPI_STMT_TYPE_MERGE");
+                break;
+            case DPI_STMT_TYPE_EXPLAIN_PLAN:
+                type = enif_make_atom(env, "DPI_STMT_TYPE_EXPLAIN_PLAN");
+                break;
+            case DPI_STMT_TYPE_COMMIT:
+                type = enif_make_atom(env, "DPI_STMT_TYPE_COMMIT");
+                break;
+            case DPI_STMT_TYPE_ROLLBACK:
+                type = enif_make_atom(env, "DPI_STMT_TYPE_ROLLBACK");
+                break;
+        }
+
+    enif_make_map_put(
+        env, map, enif_make_atom(env, "statementType"), type, &map);
```

Comment:
No need for `#{statementType => ...}` wrapping in case of single `[OUT]` parameter

Reply:
resolved